### PR TITLE
Updated for FAKE 5.1

### PR DESCRIPTION
**Pull request by zakaluka**
First attempt.

### CODE REVIEW

**Comment by zakaluka** [comment]
For #148

File: build.fsx [hunk 1]
```diff
@@ -1,58 +1,92 @@
...
+// Fake.Core.Target.create "InstallDotNetCore" (fun _ ->
+//    dotnetExePath <- DotNetCli.InstallDotNetSDK dotnetcliVersion
+// )
 
-Target "Clean" (fun _ ->
-    CleanDir "src/obj"
-    CleanDir "src/bin"
-    CleanDir "netstandard/obj"
-    CleanDir "netstandard/bin"
+Core.Target.create "Clean" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 2]
```diff
@@ -1,58 +1,92 @@
...
+    IO.Shell.cleanDir "src/obj"
+    IO.Shell.cleanDir "src/bin"
+    IO.Shell.cleanDir "netstandard/obj"
+    IO.Shell.cleanDir "netstandard/bin"
 )
 
-Target "Install" (fun _ ->
+Core.Target.create "Install" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 3]
```diff
@@ -1,58 +1,92 @@
...
     |> Seq.iter (fun s ->
         let dir = IO.Path.GetDirectoryName s
-        runDotnet dir "restore"
+        DotNet.restore (fun a -> a.WithCommon (withWorkDir dir)) s
+        // runDotnet "restore"  dir
     )
 )
 
-Target "Build" (fun _ ->
+Core.Target.create "Build" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 4]
```diff
@@ -1,58 +1,92 @@
...
+            s
+    // runDotnet dir "build -c Release /p:SourceLinkCreate=true"
+    )
 )
 
-let release = LoadReleaseNotes "RELEASE_NOTES.md"
+let release = ReleaseNotes.load "RELEASE_NOTES.md"
 
-Target "Meta" (fun _ ->
+Core.Target.create "Meta" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 5]
```diff
@@ -65,22 +99,37 @@ Target "Meta" (fun _ ->
...
+    |> IO.File.write false "Directory.Build.props"
 )
 
 // --------------------------------------------------------------------------------------
 // Build a NuGet package
 
-Target "Package" (fun _ ->
-    runDotnet "src" "pack -c Release"
-    runDotnet "netstandard" "pack -c Release"
+Core.Target.create "Package" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 6]
```diff
@@ -65,22 +99,37 @@ Target "Meta" (fun _ ->
...
+        ) s
+    )
 )
 
-Target "PublishNuget" (fun _ ->
-    let args = sprintf "nuget push Fable.Elmish.%s.nupkg -s nuget.org -k %s" (string release.SemVer) (environVar "nugetkey")
-    runDotnet "src/bin/Release" args
-    let args = sprintf "nuget push Elmish.%s.nupkg -s nuget.org -k %s" (string release.SemVer) (environVar "nugetkey")
-    runDotnet "netstandard/bin/Release" args
+Core.Target.create "PublishNuget" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 7]
```diff
@@ -170,55 +223,56 @@ Target "WatchDocs" (fun _ ->
 // --------------------------------------------------------------------------------------
 // Release Scripts
 
-Target "ReleaseDocs" (fun _ ->
+Core.Target.create "ReleaseDocs" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 8]
```diff
@@ -170,55 +223,56 @@ Target "WatchDocs" (fun _ ->
...
+    Tools.Git.Staging.stageAll tempDocsDir
+    Tools.Git.Commit.exec tempDocsDir (sprintf "Update generated documentation for version %s" release.NugetVersion)
     Branches.push tempDocsDir
 )
 
 #load "paket-files/build/fsharp/FAKE/modules/Octokit/Octokit.fsx"
 open Octokit
 
-Target "Release" (fun _ ->
+Core.Target.create "Release" (fun _ ->
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 9]
```diff
@@ -170,55 +223,56 @@ Target "WatchDocs" (fun _ ->
...
-    |> releaseDraft
+
+    Fake.Api.GitHub.createClient user pw
+    |> Fake.Api.GitHub.draftNewRelease gitOwner gitName release.NugetVersion (release.SemVer.PreRelease <> None) release.Notes
+    |> Fake.Api.GitHub.publishDraft
     |> Async.RunSynchronously
 )
 
-Target "Publish" DoNothing
+Core.Target.create "Publish" ignore
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.create`

File: build.fsx [hunk 10]
```diff
@@ -235,4 +289,4 @@ Target "Publish" DoNothing
 
 
 // start build
-RunTargetOrDefault "Build"
```
Comment (MangelMaxime) [inline build.fsx]: Please open `Fake.Core` and use `Target.runOrDefault`

File: build.fsx [hunk 11]
```diff
@@ -65,22 +99,37 @@ Target "Meta" (fun _ ->
       sprintf "<Version>%s</Version>" (string release.SemVer)
       "</PropertyGroup>"
       "</Project>"]
-    |> WriteToFile false "Directory.Build.props"
+    |> IO.File.write false "Directory.Build.props"
```
Comment (MangelMaxime) [inline build.fsx]: Prefer to use `File.write`

File: build.fsx [hunk 12]
```diff
@@ -170,55 +223,56 @@ Target "WatchDocs" (fun _ ->
...
     let tempDocsDir = "temp/gh-pages"
-    CleanDir tempDocsDir
+    Shell.cleanDir tempDocsDir
     Repository.cloneSingleBranch "" (gitHome + "/" + gitName + ".git") "gh-pages" tempDocsDir
 
-    CopyRecursive "docs/output" tempDocsDir true |> tracefn "%A"
-    StageAll tempDocsDir
-    Git.Commit.Commit tempDocsDir (sprintf "Update generated documentation for version %s" release.NugetVersion)
+    Shell.copyRecursive "docs/output" tempDocsDir true |> Trace.tracefn "%A"
+    Tools.Git.Staging.stageAll tempDocsDir
```
Comment (MangelMaxime) [inline build.fsx]: Prefer to open, `Fake.Tools.Git` and use `Staging.stageAll`

File: build.fsx [hunk 13]
```diff
@@ -170,55 +223,56 @@ Target "WatchDocs" (fun _ ->
...
+    Tools.Git.Commit.exec tempDocsDir (sprintf "Update generated documentation for version %s" release.NugetVersion)
```
Comment (MangelMaxime) [inline build.fsx]: Prefer to open, `Fake.Tools.Git` and use `Commit.exec`

File: build.fsx [hunk 14]
```diff
@@ -170,55 +223,56 @@ Target "WatchDocs" (fun _ ->
...
     let remote =
-        Git.CommandHelper.getGitResult "" "remote -v"
+        Tools.Git.CommandHelper.getGitResult "" "remote -v"
         |> Seq.filter (fun (s: string) -> s.EndsWith("(push)"))
         |> Seq.tryFind (fun (s: string) -> s.Contains(gitOwner + "/" + gitName))
         |> function None -> gitHome + "/" + gitName | Some (s: string) -> s.Split().[0]
 
-    StageAll ""
-    Git.Commit.Commit "" (sprintf "Bump version to %s" release.NugetVersion)
+    Tools.Git.Staging.stageAll ""
```
Comment (MangelMaxime) [inline build.fsx]: Prefer to open, `Fake.Tools.Git` and use `Staging.stageAll`

File: build.fsx [hunk 15]
```diff
@@ -1,58 +1,92 @@
...
+//     nuget Fake.Core >= 5.0.0
+//     nuget Fake.Core.ReleaseNotes
+//     nuget Fake.IO.FileSystem
+//     nuget Fake.Tools.Git
+//     nuget Fake.DotNet.Cli
+//     nuget Fake.DotNet.MSBuild
+//     nuget FSharp.Formatting
+//     github fsharp/FAKE modules/Octokit/Octokit.fsx
+//     //"
 #r "./packages/build/FAKE/tools/FakeLib.dll"
```
Comment (MangelMaxime) [inline build.fsx]: Prefer to use `groupref` or inlined `paket` synthax.

Favor for `groupref`:


```fs
#r "paket: groupref netcorebuild //"
#load ".fake/build.fsx/intellisense.fsx"
```

**Comment by MangelMaxime** [comment]
@zakaluka I cleaned the git history so it's easier to read and also to keep the repo history clean :)

I also tried to fix the failing travis build. Linux is case sensitive so I tried to respect that [in the load statement](https://github.com/elmish/elmish/pull/149/commits/0e6084e337f1c89c3a176383e76d6e1db8d65bb3#diff-6074983490330a0aac1bd23c958232d7R212).

It working under my mac but can't yet test it on travis because the Key server seems to failed... Will see later if travis is working again or no...

**Comment by MangelMaxime** [comment]
Ok finally travis succeeded to build the project. 

@et1975 I think the CI is green now. I am not sure why it trigger 4 CI instances and canceled 2 of them.

**Comment by et1975** [comment]
Thanks!

**Pull request merged**